### ISSUE TITLE
[Feature] bisheng编译支持O3等级

### ISSUE THREAD

**Issue by fengz72** [tile-ai/tilelang-ascend#1233]
## Feature Description
路径: `tilelang/jit/adapter/libgen.py`
现在ascendc和pto后段的编译指令都使用的 `O2` 优化等级, 导致算子编译产物未充分优化, 建议开启 `O3` 优化等级 (bisheng编译器默认等级)

## Use Case
无

## Proposed Solution
开启 `O3` 优化

## Feature Type
- [ ] New operator/kernel (e.g., new GEMM variant, new attention mechanism)
- [ ] New API/primitive (e.g., new T.xxx operation)
- [x] Compiler optimization (e.g., new pass, performance improvement)
- [ ] New backend support (e.g., new hardware target)
- [ ] Developer tooling (e.g., debugging, profiling)
- [ ] Documentation improvement
- [ ] Other: [specify]

## Operator Details (if applicable)

**Operator Type:**
[e.g. GEMM, Flash Attention, Softmax, LayerNorm, Convolution, etc.]

**Input/Output Specification:**
```
Input shapes and dtypes:
- A: [M, K], float16
- B: [K, N], float16

Output shapes and dtypes:
- C: [M, N], float16
```

**Performance Requirements:**
[e.g. target throughput, latency, memory bandwidth]

**Reference Implementation (if any):**
[e.g. PyTorch implementation, CUDA kernel, paper reference]

## Alternative Solutions
Describe any alternative solutions or features you've considered.

## Implementation Constraints (if known)

**Programming Mode Preference:**
- [ ] Developer mode (automatic)
- [ ] Expert mode (manual control)
- [x] Both modes supported

**Memory Constraints:**
[e.g. L1 buffer size, UB buffer size constraints]

**Target Hardware:**
- [x] Ascend A2
- [x] Ascend A3
- [ ] Other Ascend variants

## Additional Context
Add any other context, references, or screenshots about the feature request here.

## Willingness to Contribute
- [x] I'm willing to submit a PR for this feature
- [ ] I can help with testing
- [ ] I can provide reference implementations
- [ ] I need guidance to get started

**Comment by fengz72** [comment]
已在中修复 https://github.com/tile-ai/tilelang-ascend/issues/1302
